libs/core/langchain_core/indexing/api.py

```diff
@@ -198,6 +198,7 @@ def index(
     source_id_key: Union[str, Callable[[Document], str], None] = None,
     cleanup_batch_size: int = 1_000,
     force_update: bool = False,
+    **kwargs: Any,
```

Comment:
Don't use `**` here to avoid confusion of which namespace kwargs belong (e.g., what if we need kwargs for delete at some point or else add other parameters to index in the future).

Instead can call is `upsert_kwargs`. Please document which API whey will be passed to in the doc-string

Reply:
Thank you for the feedback. I've implemented the changes as suggested:

1. Replaced `**kwargs` with a specific `upsert_kwargs: Optional[dict[str, Any]] = None` parameter.
2. Updated the docstring to clarify that these kwargs will be passed to the `add_documents` and `aadd_documents` method of the VectorStore.
3. Added an example in the docstring to demonstrate how `upsert_kwargs` can be used, specifically mentioning the `vector_field` use case.

Reply:
Also added some unit tests for `index` and `aindex` with `upsert_kwargs`.